packages/raystack/calendar/range-picker.tsx

```diff
@@ -16,31 +16,38 @@ interface RangePickerProps {
   calendarProps?: CalendarProps;
   onSelect?: (date: DateRange) => void;
   pickerGroupClassName?: string;
+  value?: DateRange;
 }
 
+type RangeFields = keyof DateRange;
+
 export function RangePicker({
   side = "top",
   dateFormat = "DD/MM/YYYY",
   textFieldProps,
   calendarProps,
   onSelect = () => {},
+  value = {
+    to: new Date(),
+    from: new Date(),
+  },
   pickerGroupClassName,
 }: RangePickerProps) {
   const [showCalendar, setShowCalendar] = useState(false);
-  const [value, setValue] = useState<DateRange>({
-    to: new Date(),
-    from: new Date(),
-  });
+  const [currentRangeField, setCurrentRangeField] =
+    useState<RangeFields>("from");
   const startDate = dayjs(value.from).format(dateFormat);
   const endDate = dayjs(value.to).format(dateFormat);
 
   const handleSelect: SelectRangeEventHandler = (range, selectedDay) => {
-    if (range) {
-      setValue(range);
-    }
-    if (range?.to === selectedDay) {
-      onOpenChange(false);
-      onSelect(range);
+    let from = value?.from || range?.from;
+    from = dayjs(from).isAfter(dayjs(selectedDay)) ? undefined : from;
```

Comment:
Couldn't quite understand the logic from 43-50. I'm assuming it has to do with "auto-populating" to and from depending on what the user has clicked. Can you explain the expected behaviour we are targeting once?

Thinking out loud if there is scope to make this more readable/understandable

Reply:
I added comments and simplified the logic here.